docs/tutorials/09-complete-tutorials/04-zkverify-js.md

```diff
@@ -0,0 +1,240 @@
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+In this tutorial we will be verifying proofs using zkVerify JS package. ```zkVerify JS``` is a NPM package which makes it very easy to submit proofs, listen events and get aggregation proofs. You can use this package with all the proof types we support.
+
+Let's create a new project and install ```zkverify JS``` for our project. Run the following commands:
+```bash
+# Creating a new directory
+mkdir proof-submission
+
+# Navigating to the project directory
+cd proof-submission
+
+# Initializing an NPM project
+npm init
+
+# Installing zkVerify JS
+npm i zkverifyjs
+```
+
+Create a new file named ```index.js``` to write the verification logic. Open ```index.js``` in your IDE and import the following components from ```zkVerify JS``` :
+
+<Tabs groupId="import">
+<TabItem value="circom" label="Circom">
+```js
+const {zkVerifySession, Library, CurveType, ZkVerifyEvents} = require("zkverifyjs");
+```
+</TabItem>
+<TabItem value="r0" label="Risc Zero">
+```js
+const {zkVerifySession, ZkVerifyEvents} = require("zkverifyjs");
+```
+</TabItem>
+<TabItem value="noir" label="Noir">
+```js
+const {zkVerifySession, ZkVerifyEvents} = require("zkverifyjs");
+```
+</TabItem>
+</Tabs>
+
+We would also need to import the required files we have generated already in previous tutorials, which are proof, verification key and public inputs. Use the following code snippets :
+
+<Tabs groupId="import-files">
+<TabItem value="circom" label="Circom">
+```js
+const fs = require("fs");
+const proof = require("./data/proof.json");
+const public = require("./data/public.json");
+const key = require("./data/main.groth16.vkey.json");
+```
+</TabItem>
+<TabItem value="r0" label="Risc Zero">
+```js
+const fs = require("fs");
+const proof = require("../my_project/proof.json"); // Following the Risc Zero tutorial
+```
+</TabItem>
+<TabItem value="noir" label="Noir">
+```js
+const fs = require("fs");
+const proof = fs.readFileSync("../hello_world/proof.hex").toString()
+const public = fs.readFileSync("../hello_world/pub.hex").toString()
+```
+</TabItem>
+</Tabs>
+
+Once you have all the requirements imported, we will start by instantiating a session with our Volta testnet with an account(This account should have $tVFY to pay for transactions). 
+```js
+const session = await zkVerifySession.start().Volta().withAccount("seed-phrase")
+```
+
+For Circom and Noir proofs, we need to register a verification key on our Volta testnet which will be used in further steps while verifying proofs. This step is not required for Risc Zero because we already got a hash of the image id which can be used directly. You can execute the following code snippet to register a vkey:
+
+<Tabs groupId="register-vkey">
+<TabItem value="circom" label="Circom">
+```js
+const {events, regResult} = await session.registerVerificationKey().groth16({library: Library.snarkjs, curve: CurveType.bn128}).execute(key);
+
```

Comment:
Might be worth specifying that it's possible to use also the [getVkHash](https://github.com/zkVerify/zkverifyjs?tab=readme-ov-file#zkverifysessiongetvkhash) function to retrieve the hash of the vk as formatted by the zkVerify chain.